gluonnlp/data/sampler.py

```diff
@@ -207,16 +207,14 @@ def __init__(self, lengths, batch_size, num_buckets=10, bucket_keys=None,
                                    if len(sample_ids) > 0]
         scale_up_keys = [key if self._single_element else sum(key) for key in self._bucket_keys]
         max_scale_up_key = max(scale_up_keys)
-        if batch_type == 'sentence':
+        if ~use_average_length:
```

Comment:
`if not use_avg_length`

Reply:
also, the scale_up_keys only needs to be calculated in this case, so you can move those logic into this `if` block.